File: Cosmos-IOS/Cosmostation/Cosmostation/Model/KavaModel/IncentiveParam.swift
//
//  IncentiveParam.swift
//  Cosmostation
//
//  Created by 정용주 on 2021/03/04.
//  Copyright © 2021 wannabit. All rights reserved.
//

import Foundation

public struct IncentiveParam {
    var claim_multipliers: Array<ClaimMultiplier> = Array<ClaimMultiplier>()
    
    init(_ dictionary: NSDictionary?) {
        if let rawClaimMultipliers = dictionary?["claim_multipliers"] as? Array<NSDictionary>  {
            for rawClaimMultiplier in rawClaimMultipliers {
                self.claim_multipliers.append(ClaimMultiplier(rawClaimMultiplier))
            }
        }
    }
    
    public func getFactor(_ denom: String, _ position: Int) -> NSDecimalNumber{
        for claim_multipliers in claim_multipliers {
            if (claim_multipliers.denom == denom) {
                return claim_multipliers.multipliers[position].factor
            }
        }
        return NSDecimalNumber.zero
    }
}

File: app/src/main/java/wannabit/io/cosmostaion/model/kava/IncentiveParam.java
package wannabit.io.cosmostaion.model.kava;

import com.google.gson.annotations.SerializedName;

import java.math.BigDecimal;
import java.util.ArrayList;

public class IncentiveParam {
    @SerializedName("claim_multipliers")
    public ArrayList<ClaimMultiplier> claim_multipliers;

    public BigDecimal getFactor(String denom, int position) {
        for (ClaimMultiplier claimMultiplier : claim_multipliers) {
            if (claimMultiplier.denom.equalsIgnoreCase(denom)) {
                return new BigDecimal(claimMultiplier.multipliers.get(position).factor);
            }
        }
        return BigDecimal.ZERO;
    }
}
